src/Microsoft.AspNetCore.Mvc.Core/ProducesAttribute.cs

```diff
@@ -87,7 +87,19 @@ public override void OnResultExecuting(ResultExecutingContext context)
             {
                 // Check if there are any IFormatFilter in the pipeline, and if any of them is active. If there is one,
                 // do not override the content type value.
-                if (context.Filters.OfType<IFormatFilter>().All(f => f.GetFormat(context) == null))
+                var filtersInactive = true;
```

Comment:
An alternative might be to `return` from the `for` loop, instead of a break (which avoids the need to define this variable). It does seem ever so slightly harder to read though.